File: snippets/dart_snippets/lib/static_channel_backup.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:breez_sdk/bridge_generated.dart';

Future<StaticBackupResponse> retrieveBackupFiles({required String workingDir}) async {
  // ANCHOR: static-channel-backup
  StaticBackupRequest req = StaticBackupRequest(workingDir: workingDir);
  StaticBackupResponse resp = await breezSDK.staticBackup(req: req);
  // ANCHOR_END: static-channel-backup
  return resp;
}

File: snippets/swift/BreezSDKExamples/Sources/StaticChannelBackup.swift
//
//  StaticChannelBackup.swift
//  BreezSDKDocs
//
//  Created by ruben on 14/11/2023.
//

import BreezSDK
import Foundation

func retrieveBackupFiles() -> StaticBackupResponse? {
    // ANCHOR: static-channel-backup
    let backupData = try? staticBackup(req: StaticBackupRequest(workingDir: "<working directory>"))
    // ANCHOR_END: static-channel-backup
    return backupData
}
